packages/instrumentation/src/ActiveTracingHelper.ts

```diff
@@ -42,42 +44,59 @@ export class ActiveTracingHelper implements TracingHelper {
   }
 
   createEngineSpan(engineSpanEvent: EngineSpanEvent) {
+    const rootEngineSpans = ['prisma:engine', 'prisma:engine:itx_runner']
+
+    for (const span of engineSpanEvent.spans) {
+      // We receive engine spans as soon as they end. So the parent nodes are received after the children.
+      // Because of that we cannot process the spans immediately, as creating spans affects them a random SpanId and that
+      // breaks the `parent_span_id` relationship.
+      // To circumvent this issue we accumulate engine spans and only process them once a full span tree is acquired.
+      const currentValue = this.engineSpansToProcess.get(span.parent_span_id) ?? []
+      this.engineSpansToProcess.set(span.parent_span_id, [...currentValue, span])
```

Comment:
To circumvent the SpanId problem, one way would be to accumulate traces and process them once we know we have a full "tree". 
Some notes:
- are spans always nested in either prisma:engine or prisma:engine:itx_runner ?
- spans should arrive in order, but who knows
- if a root span never arrives for some reason it could mean a small memory leak. A regular cleanup of old & unused spans might be needed

Anyway, I know it might not be an acceptable solution for you guys, but I just wanted to propose an alternative that could work on paper - or at least pass the tests :)

Reply:
Any thoughts on whether or not this would be an acceptable solution @SevInf ? If it's not I will close the PR.